File: app/src/main/java/com/example/vu/android/empowerplant/AppDatabase.java
package com.example.vu.android.empowerplant;


import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {StoreItem.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {
    public abstract StoreItemDAO StoreItemDAO();

    private static volatile AppDatabase INSTANCE;
    //singleton recommended
    public static synchronized AppDatabase getInstance(Context context){
        if (INSTANCE == null){
            INSTANCE = Room.databaseBuilder(context.getApplicationContext(), 
            AppDatabase.class,"AppDatabase")
            .allowMainThreadQueries()
            .build();
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/vu/android/empowerplant/db/AppDatabase.kt
package com.example.vu.android.empowerplant.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [StoreItem::class], version = 2, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun StoreItemDAO(): StoreItemDAO

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        // singleton recommended
        fun getInstance(context: Context): AppDatabase {
            if (INSTANCE == null) {
                synchronized(AppDatabase::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room
                            .databaseBuilder(
                                context.applicationContext,
                                AppDatabase::class.java, "db"
                            )
                            .build()
                    }
                }
            }
            return INSTANCE!!
        }
    }
}